src/emd/rt_bse_io.F

```diff
@@ -0,0 +1,807 @@
+      n = rtbse_env%sim_nsteps + 2
+      NULLIFY (omega_series)
+      ALLOCATE (omega_series(n), source=0.0_dp)
+      NULLIFY (ft_real_series)
+      ALLOCATE (ft_real_series(n), source=0.0_dp)
+      NULLIFY (ft_imag_series)
+      ALLOCATE (ft_imag_series(n), source=0.0_dp)
+      NULLIFY (value_series)
+      ALLOCATE (value_series(n), source=0.0_dp)
+      ALLOCATE (ft_full_series(6, n))
+      ! Carry out for each direction independently and real and imaginary parts also independently
+      DO i = 1, 3
+         ! Real part of the value first
+         value_series(:) = REAL(rtbse_env%moments_trace(i)%series(:))
+         CALL ft_simple(rtbse_env%time_trace%series, value_series, omega_series, ft_real_series, ft_imag_series, &
+                        rtbse_env%ft_damping, rtbse_env%ft_start)
+         ft_full_series(2*i - 1, :) = ft_real_series(:)
+         ft_full_series(2*i, :) = ft_imag_series(:)
+         ! Now imaginary part
+         value_series(:) = AIMAG(rtbse_env%moments_trace(i)%series(:))
+         CALL ft_simple(rtbse_env%time_trace%series, value_series, omega_series, ft_real_series, ft_imag_series, &
+                        rtbse_env%ft_damping, rtbse_env%ft_start)
+         ft_full_series(2*i - 1, :) = ft_full_series(2*i - 1, :) - ft_imag_series
+         ft_full_series(2*i, :) = ft_full_series(2*i, :) + ft_real_series
+      END DO
+      DEALLOCATE (ft_real_series)
+      DEALLOCATE (ft_imag_series)
+      DEALLOCATE (value_series)
+      ! Now, write these to file
+      ft_unit = cp_print_key_unit_nr(logger, rtbse_env%ft_section, extension=".dat", &
+                                     file_form="FORMATTED", file_position="REWIND")
+      IF (ft_unit > 0) THEN
+         DO i = 1, n
+            WRITE (ft_unit, '(E20.8E3,E20.8E3,E20.8E3,E20.8E3,E20.8E3,E20.8E3,E20.8E3)') &
+               omega_series(i), ft_full_series(1, i), ft_full_series(2, i), &
+               ft_full_series(3, i), ft_full_series(4, i), &
+               ft_full_series(5, i), ft_full_series(6, i)
+         END DO
+      END IF
+      CALL cp_print_key_finished_output(ft_unit, logger, rtbse_env%ft_section)
+      DEALLOCATE (omega_series)
+      DEALLOCATE (ft_full_series)
+   END SUBROUTINE output_moments_ft
+! **************************************************************************************************
+!> \brief Outputs the isotropic polarizability tensor element alpha _ ij = mu_i(omega)/E_j(omega),
+!>        where i and j are provided by the configuration. The tensor element is energy dependent and
+!>        has real and imaginary parts
+!> \param rtbse_env GW-BSE environment
+! **************************************************************************************************
+   SUBROUTINE output_polarizability(rtbse_env)
+      TYPE(rtbse_env_type), POINTER                      :: rtbse_env
+      TYPE(cp_logger_type), POINTER                      :: logger
+      REAL(kind=dp), DIMENSION(:), POINTER               :: omega_series, &
```

Comment:
Does this have to be a pointer or could an allocatable do the job as well?

Reply:
I changed the type to `ALLOCATABLE`.